### PR TITLE
docs: add documentation from boost-histogram

### PR DESCRIPTION
**Pull request by amangoel185**
In reference to #265

### CODE REVIEW

File: docs/user-guide/comparison.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.. _usage-comparison:
+
+Comparison with Boost.Histogram
```
Comment (henryiii) [inline docs/user-guide/comparison.rst:3]: We might want to just link to this page, remove it, or at least prefix it with a comparison to boost-histogram. Users will not be as interested in Boost.Histogram here as they would be in a library called boost-histogram. A page hist vs. boost-histogram is almost what we have in the readme.

File: docs/user-guide/indexing.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+.. _usage-indexing:
+
+Indexing
+========
+
+Hist implements the UHI indexing protocol. You can read more about it on the `UHI Indexing <https://uhi.readthedocs.io/en/latest/indexing.html>`_ page.
```
Comment (henryiii) [inline docs/user-guide/indexing.rst:6]: We could also link to the UHI+ page that follows, since Hist follows that one too.

**Comment by henryiii** [comment]
Looks good minus a couple of small comments.

File: docs/user-guide/comparison.rst (hunk 1 above)
Comment (amangoel185) [reply docs/user-guide/comparison.rst:3]: Alright! Should we show both the comparisons on the same page or different ones?

File: docs/user-guide/indexing.rst (hunk 2 above)
Comment (amangoel185) [reply docs/user-guide/indexing.rst:6]: Okay

File: docs/user-guide/comparison.rst (hunk 1 above)
Comment (henryiii) [reply docs/user-guide/comparison.rst:3]: I'd not bother with repeating the comparisons in Hist, users can go to boost-histogram for comparisons with Boost.Histogram. I'd just not have this page at all, I think.

Comment (amangoel185) [reply docs/user-guide/comparison.rst:3]: Okay, that makes sense!

**Review by henryiii — APPROVED** [review]

**Pull request merged**